clients/lsp-clojure.el

```diff
@@ -46,12 +46,17 @@
 
 (defcustom lsp-clojure-server-download-url
   (format "https://github.com/clojure-lsp/clojure-lsp/releases/latest/download/clojure-lsp-native-%s.zip"
-          (pcase system-type
-            ('gnu/linux "linux-amd64")
-            ('darwin (if (string-match "^aarch64-.*" system-configuration)
-                         "macos-aarch64"
-                       "macos-amd64"))
-            ('windows-nt "windows-amd64")))
+          (let ((arch (substring system-configuration 0 (string-search "-" system-configuration))))
```

Comment:
nit: there's a new value `lsp--system-arch` that can be used with `(lsp-resolve-value lsp--system-arch)` to get the architecture of Emacs

Reply:
https://github.com/emacs-lsp/lsp-mode/blob/master/clients/lsp-lua.el#L545-L547
I'll merge the PR to unblock the issue, feel free to open a new PR